### PR TITLE
:seedling: compare status update event sequence ID in source client

### PR DESCRIPTION
**Pull request by skeeey**
## Summary
Compare status update event sequence ID in work source client to ensure that the messages received by the source client are in the same order as the messages sent by the agent client

This will help to guard against if some message queues cannot guarantee the order of the messages they send

### CODE REVIEW

File: pkg/cloudevents/work/store/base.go [hunk 1]
```diff
@@ -200,13 +200,31 @@ func (b *workProcessor) handleWork(work *workv1.ManifestWork) error {
...
 	}
 
+	if updatedWork.Annotations == nil {
+		updatedWork.Annotations = map[string]string{}
+	}
+	lastSequenceID := lastWork.Annotations[common.CloudEventsSequenceIDAnnotationKey]
+	sequenceID := work.Annotations[common.CloudEventsSequenceIDAnnotationKey]
+	greater, err := utils.CompareSnowflakeSequenceIDs(lastSequenceID, sequenceID)
+	if err != nil {
+		klog.Errorf("invalid resource version for work %s/%s, %v", lastWork.Namespace, lastWork.Name, err)
```
Comment (qiujian16) [inline pkg/cloudevents/work/store/base.go:210]: s/invalid resource version/invalid sequenceID

Comment (skeeey) [reply pkg/cloudevents/work/store/base.go]: done

**Pull request merged**